cmd/promoted-image-governor/main.go

```diff
@@ -399,16 +454,37 @@ func main() {
 		return
 	}
 
-	kubeConfig, err := clientcmd.BuildConfigFromFlags("", opts.kubeconfig)
+	kubeconfigs, err := opts.kubernetesOptions.LoadClusterConfigs()
 	if err != nil {
-		logrus.WithError(err).Fatalf("could not load kube config from path %s", opts.kubeconfig)
+		logrus.WithError(err).Fatal("failed to load kubeconfigs")
+	}
+	if _, hasAppCi := kubeconfigs[appCIContextName]; !hasAppCi {
+		InClusterConfig, err := rest.InClusterConfig()
```

Comment:
See first how this goes.
https://github.com/openshift/ci-tools/pull/2354#discussion_r704762206